Packages not generated due to an error in the publish-alerting stage

## Description

While analyzing https://github.com/wazuh/wazuh-qa-automation/issues/6632, it has been detected that the indexer packages were not generated due to the following failure:

```
Caused by: org.gradle.internal.resolve.ModuleVersionNotFoundException: Could not find org.opensearch:common-utils:5.0.0.0.
Searched in the following locations:
  - file:/home/runner/.m2/repository/org/opensearch/common-utils/5.0.0.0/common-utils-5.0.0.0.pom
  - https://ci.opensearch.org/ci/dbc/snapshots/maven/org/opensearch/common-utils/5.0.0.0/common-utils-5.0.0.0.pom
  - https://repo.maven.apache.org/maven2/org/opensearch/common-utils/5.0.0.0/common-utils-5.0.0.0.pom
Required by:
    project ':alerting-core'


BUILD FAILED in 10s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.2.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
5 actionable tasks: 5 executed
Error: Process completed with exit code 1.
```

## Evidence

- Builds:
   - https://github.com/wazuh/wazuh-indexer/actions/runs/24540637634/job/71746920981
   - https://github.com/wazuh/wazuh-indexer/actions/runs/24540636951/job/71746783959
   - https://github.com/wazuh/wazuh-indexer/actions/runs/24540633692/job/71746503397
   - https://github.com/wazuh/wazuh-indexer/actions/runs/24540633142/job/71746405576

--- a/.github/workflows/5_builderpackage_security-analytics.yml
+++ b/.github/workflows/5_builderpackage_security-analytics.yml
@@ -105,7 +105,7 @@ jobs:
 
   publish-alerting:
     runs-on: ubuntu-24.04
-    needs: [ branches, get-version ]
+    needs: [ branches, get-version, publish-common-utils ]
     steps:
       - name: Publish alerting to Maven Local
         uses: wazuh/wazuh-indexer-alerting/.github/actions/5_local_maven_publisher@main

--- a/.github/workflows/5_builderpackage_security-analytics.yml
+++ b/.github/workflows/5_builderpackage_security-analytics.yml
@@ -107,6 +107,11 @@ jobs:
     runs-on: ubuntu-24.04
     needs: [ branches, get-version, publish-common-utils ]
     steps:
+      - name: Restore Maven Local cache (common-utils)
+        uses: actions/cache/restore@v5
+        with:
+          path: ~/.m2/repository
+          key: maven-common-utils-${{ github.ref_name }}
       - name: Publish alerting to Maven Local
         uses: wazuh/wazuh-indexer-alerting/.github/actions/5_local_maven_publisher@main
         with:

